src/micromamba/clean.cpp

```diff
@@ -19,24 +19,32 @@ init_clean_parser(CLI::App* subcom)
 {
     init_general_options(subcom);
 
-    subcom->add_flag("-a,--all",
-                     clean_options.all,
-                     "Remove index cache, lock files, unused cache packages, and tarballs.");
-    subcom->add_flag("-i,--index-cache", clean_options.index_cache, "Remove index cache.");
-    subcom->add_flag("-p,--packages",
-                     clean_options.packages,
-                     "Remove unused packages from writable package caches.\n"
-                     "WARNING: This does not check for packages installed using\n"
-                     "symlinks back to the package cache.");
-    subcom->add_flag("-t,--tarballs", clean_options.tarballs, "Remove cached package tarballs.");
-}
+    auto& config = Configuration::instance();
 
-void
-load_clean_options(Context& ctx)
-{
-    load_general_options(ctx);
-    load_prefix_options(ctx);
-    load_rc_options(ctx);
+    auto& clean_all = config.insert(
```

Comment:
just for me to understand: this is technically slightly over-engineered since these options are only ever set through CLI, right? We could also just keep the old way here, right?

Reply:
Indeed.
I tried not to mix different approaches to have a single way to define and use configuration.
It also gives you capability to turn on rc configuration or more useful here, config by environment variable in one line : `.set_env_var_name()`

Reply:
it will also make much more sense when moving the high-level API of `mamba`: `install`, `update`, `clean`, etc. from `micromamba` to `mamba`/`libmamba` itelf. Because you will maybe want to be able to configure these using rc files (`clean_all: true` in rc file makes `clean` command always cleaning packages/tarballs/index)